apps/css-workshop/src/components/Icon.astro

```diff
@@ -0,0 +1,43 @@
+---
+type Props = {
+  svg: string;
+  size?: 's' | 'm' | 'l' | 'auto';
+  wrapperProps?: astroHTML.JSX.HTMLAttributes;
+  buttonIcon?: boolean;
+} & astroHTML.JSX.HTMLAttributes;
+
+const {
+  svg,
+  size = 'm',
+  class: className,
+  buttonIcon = false,
+  wrapperProps,
+  ...rest
+} = Astro.props;
+
+const iconProps = {
+  ['class:list']: [
+    {
+      ['iui-svg-icon']: !buttonIcon,
+      ['iui-button-icon']: buttonIcon,
+    },
+    className,
+  ],
+  ['data-iui-icon-size']: size,
+  ['aria-hidden']: 'true',
```

Comment:
Any wrapping component should provide `aria-hidden` to the icon as well, like #2012.

Reply:
fixed this in my last push

Reply:
You shouldn't include this `aria-hidden` as a constant value in the `iconProps`, since only `basic` icons that aren't being placed in components like `IconButton` that wrap with `aria-hidden` will need it.

Reply:
@Ben-Pusey-Bentley, can you confirm this is fixed and as expected?  `basic` is no longer a variable, https://github.com/iTwin/iTwinUI/pull/2019#discussion_r1600177777.

Reply:
It looks like the issue should be resolved, since the `aria-hidden` constant value has been removed. I haven't been able to test, though, because I haven't been able to get the branch to build locally. The CI checks also do not seem to be running correctly, so I can't currently check through deploy preview. Maybe solving the merge conflicts will help?

Reply:
Merge conflicts resolved.

Reply:
Able to test now. Looks good!